### PR TITLE
Minter keeps track of minted bonds for instance check

### PR DESCRIPTION
**Pull request by aalavandhan**
Downstream services which accept bonds from a given minter need to be able to verify if a given bond was indeed minted using the minter. 

Minter keeps track of all the bond addresses minted ..

### CODE REVIEW

File: test/BondMinter.ts [hunk 1]
```diff
@@ -104,8 +104,14 @@ describe("BondConfigVault", () => {
 
       await time.setNextBlockTimestamp(timestampAfter);
 
-      await bondMinter.mintBonds();
-
+      expect(await bondMinter.mintBonds())
```
Comment (marktoda) [inline test/BondMinter.ts:107]: await has to be outside the expect to assert events

File: test/BondMinter.ts [hunk 2]
```diff
@@ -122,15 +128,25 @@ describe("BondConfigVault", () => {
 
       await time.setNextBlockTimestamp(timestampAfter);
 
-      await bondMinter.mintBonds();
+      expect(await bondMinter.mintBonds())
```
Comment (marktoda) [inline test/BondMinter.ts:131]: ditto

File: test/BondMinter.ts [hunk 3]
```diff
@@ -122,15 +128,25 @@ describe("BondConfigVault", () => {
...
+      expect(await bondMinter.isInstance("0x0000000000000000000000000000000000000002")).to.be.false;
 
       await mockBondFactory.mock.createBond
         .withArgs(mockUnderlyingToken.address, [100, 200, 700], timestampAfter + DAYS_30 + 100)
         .returns("0x0000000000000000000000000000000000000002");
 
       await time.setNextBlockTimestamp(timestampAfter + DAYS_30);
 
-      await bondMinter.mintBonds();
+      expect(await bondMinter.mintBonds())
```
Comment (marktoda) [inline test/BondMinter.ts:144]: ditto

File: test/BondMinter.ts (hunk 1 above)
Comment (aalavandhan) [reply test/BondMinter.ts]: ah. I was curious why the linter was complaining

**Review by marktoda — APPROVED** [review]

**Pull request merged**